CONTRIBUTING.md

```diff
@@ -7,6 +7,101 @@
+interactions with the project.
+
+Contributions to this project should conform to the [Developer Certificate of
+Origin][dco].  See the [next section](#sign-your-work) for more details.
+
+## Sign Your Work
+
+Contributions to this project should conform to the [Developer Certificate of
+Origin][dco].  You need to sign-off your git commits before sending the pull
+requests.  The sign-off is a single line of text at the end of the commit
+message.  The signature consists of your official name and email address.  These
+two details should match with the name and email address used in the Git commit.
+All your commits needs to be signed.  Your signature certifies that you wrote
+the patch or otherwise have the right to contribute the material.  The rules are
+pretty simple, if you can certify the below (from
+[developercertificate.org][dco]):
+
+```
+Developer Certificate of Origin
+Version 1.1
+
+Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
+1 Letterman Drive
+Suite D4700
+San Francisco, CA, 94129
+
+Everyone is permitted to copy and distribute verbatim copies of this
+license document, but changing it is not allowed.
+
+
+Developer's Certificate of Origin 1.1
+
+By making a contribution to this project, I certify that:
+
+(a) The contribution was created in whole or in part by me and I
+    have the right to submit it under the open source license
+    indicated in the file; or
+
+(b) The contribution is based upon previous work that, to the best
+    of my knowledge, is covered under an appropriate open source
+    license and I have the right under that license to submit that
+    work with modifications, whether created in whole or in part
+    by me, under the same open source license (unless I am
+    permitted to submit under a different license), as indicated
+    in the file; or
+
+(c) The contribution was provided directly to me by some other
+    person who certified (a), (b) or (c) and I have not modified
+    it.
+
+(d) I understand and agree that this project and the contribution
+    are public and that a record of the contribution (including all
+    personal information I submit with it, including my sign-off) is
+    maintained indefinitely and may be redistributed consistent with
+    this project or the open source license(s) involved.
+```
+
+Then you just add a line to every git commit message:
+
+    Signed-off-by: Joe Smith <joe.smith@example.com>
+
+Use your real name (sorry, no pseudonyms or anonymous contributions.)
+
+If you set your `user.name` and `user.email` git configs, you can sign your
+commit automatically with `git commit -s`.
+
+Note: If your git config information is set properly then viewing the `git log`
+information for your commit will look something like this:
+
+```
+Author: Joe Smith <joe.smith@example.com>
+Date:   Thu Feb 2 11:41:15 2018 -0800
+
+    Update README
+
+    Signed-off-by: Joe Smith <joe.smith@example.com>
```

Comment:
we should add CI job checking this, preferably in this PR.

Reply:
Let's see, if we can enable this: https://probot.github.io/apps/dco/

Reply:
it looks it is configured now?

Reply:
Yes. See this new PR #834